pkg/imgpkg/bundle/contents.go

```diff
@@ -43,13 +43,14 @@ func NewContents(paths []string, excludedPaths []string, preservePermissions boo
 }
 
 // Push the contents of the bundle to the registry as an OCI Image
-func (b Contents) Push(uploadRef regname.Tag, registry ImagesMetadataWriter, logger Logger) (string, error) {
+func (b Contents) Push(uploadRef regname.Tag, labels map[string]string, registry ImagesMetadataWriter, logger Logger) (string, error) {
 	err := b.validate()
 	if err != nil {
 		return "", err
 	}
 
-	labels := map[string]string{BundleConfigLabel: "true"}
+	labels[BundleConfigLabel] = "true"
```

Comment:
I have a check for that in the PushOptions `Run()` method in the [/pkg/imgpkg/cmd/push.go](https://github.com/phenixblue/imgpkg/blob/e03fe72406ac4f3b634395a727a9a5ccb25fce20/pkg/imgpkg/cmd/push.go#L72) file that gets called before either `pushBundle()` or `pushImage` do, which is the caller to this area of code. Do you think it's better to explicitly have the check in each `Push()` method for Images and Bundles?

Reply:
My main concern is that we assume that we can just update the variable in this particular place, so I would prefer the check to live at this level. The next time `labels` is used is inside NewFileImage function, and we just check for `len` of it, and if it is `nil` it will not fail.

Reply:
OK. I will move the check for nil map to here.